### ISSUE TITLE
Issue with CJS default exports

### ISSUE THREAD

**Issue by santimirandarp** [mljs/regression-base#17]
I've tested the exports from this package which fail using `require` since the default exports in TS are translated to `exports.default =  BaseRegression` when compiling the code to CJS.

The imports work well using the `import` syntax, so Typescript or esm modules seem to be fine.

Using named exports will cause updates upstream since it will be breaking.

I'm just raising the issue so you may decide or discuss what is the best option.

The only way to keep the same syntax (no named exports) that I find (I think) is to export 2 different index files, one goes to module, one to CommonJS.

**Comment by targos** [comment]
Yeah, a breaking change in this module would be annoying for little benefit.

> The only way to keep the same syntax (no named exports) that I find (I think) is to export 2 different index files, one goes to module, one to CommonJS.

Here's my suggestion:
- Keep `src`, and TS compilation the same.
- Add a new CJS file at the root of the repo which does `module.exports = require('./lib/index.js').default`
- Add this file to the `files` array in package.json and set it as the `main` field.

**Comment by santimirandarp** [comment]
This is great, thank you. I have noticed that there is a minor bug as well, so I'll quickly submit all, and maybe we can release it.

**Comment by santimirandarp** [comment]
I've tried this for a while using something like this (since we have more exports in index):

```javascript
// Purpose: entry point for NodeJS, json.main field.
// this file is ignored by the TS compiler (tsconfig.json)
const allExports = require('./lib/index.js')

module.exports = allExports.default
exports.checkArrayLength = allExports.checkArrayLength
exports.maybeToPrecision = allExports.maybeToPrecision
```

There is the same error, however, I realised that in this case the `js` code runs fine, the error is just Typescript.

I'm a bit confused, but [I did this mini example](
https://www.typescriptlang.org/play?downlevelIteration=true&target=7&jsx=0&module=6&allowSyntheticDefaultImports=true&ssl=5&ssc=2&pln=3&pc=1#code/FAehAIGMHsFtYKYDsAu4DuALZ4UGcYkAzASwHNwBXPBPcW2YBADwAdoAnNAEwSIENKAGzSQh-PHQCyATwDC4yQEFwAb2DAAvhrBQ4iVBmxJcBaMXJUadSACs8oCC3ZdwAXiiLp8rwCE14Nrg4EA) in case you want to take a quick look.

So we may need to either trick the compiler, or use two separate ts files (I didn't test this yet.)

**Comment by targos** [comment]
> the error is just Typescript.

What does that mean? 

Your code isn't right, though. When you do `module.exports = allExports.default`, you override the default `exports` object, so you're not supposed to assign to it afterwards. I suggest:

```
const allExports = require('./lib/index.js');

module.exports = allExports.default;
Object.assign(module.exports, allExports);
```

**Comment by santimirandarp** [comment]
Thanks for the correction. Just to answer the question, by that line I mean this:

![Screenshot at 2023-10-07 20-27-11](https://github.com/mljs/regression-base/assets/29411936/a8f4f56e-0d7f-4946-b16c-50b0480ea59f)
